ci/unit_tests/functions_deploy/main_test.py

```diff
@@ -0,0 +1,198 @@
+        functionNames = self._getFunctionsInPackage(self.package)
+
+        # get original list of cloud function files and check if all of them were uploaded
+        functionsDir = os.path.join(self.dataBasePath, 'example_functions')
+        functionFileNames = [os.path.splitext(fileName)[0] for fileName in os.listdir(functionsDir)]
+        assert set(functionNames) == set(functionFileNames)
+
+        # try to call particular functions
+        for functionName in functionNames:
+            functionCallUrl = self.cloudFunctionsUrl + '/' + self.urlNamespace + '/actions/' + self.package + \
+                '/' + functionName + '?blocking=true&result=true'
+
+            functionResp = requests.post(functionCallUrl, auth=(os.environ['CLOUD_FUNCTIONS_USERNAME'],
+                                                                os.environ['CLOUD_FUNCTIONS_PASSWORD']),
+                                         headers={'Content-Type': 'application/json',
+                                                  'accept': 'application/json'},
+                                         data=json.dumps({'name': 'unit test'}))
+
+            assert functionResp.status_code == 200
+            functionRespJson = functionResp.json()
+            assert "Hello unit test!" in functionRespJson['greeting']
+
+
+    def test_pythonVersionFunctions(self):
+        """Tests if it's possible to upload one fuction into two different version of runtime."""
+        for pythonVersion in [2, 3]:
+            params = ['-c', os.path.join(self.dataBasePath, 'python' + str(pythonVersion) + 'Functions.cfg'),
+                      '--cloudfunctions_username', os.environ['CLOUD_FUNCTIONS_USERNAME'],
+                      '--cloudfunctions_password', os.environ['CLOUD_FUNCTIONS_PASSWORD'],
+                      '--cloudfunctions_package', self.package, '--cloudfunctions_namespace', self.urlNamespace,
+                      '--cloudfunctions_url', self.cloudFunctionsUrl]
+
+            self.t_noException([params])
+            self.packageCreated = True
+
+            functionCallUrl = self.cloudFunctionsUrl + '/' + self.urlNamespace + '/actions/' + self.package + \
+                '/getPythonMajorVersion?blocking=true&result=true'
+
+            functionResp = requests.post(functionCallUrl, auth=(os.environ['CLOUD_FUNCTIONS_USERNAME'],
+                                                                os.environ['CLOUD_FUNCTIONS_PASSWORD']),
+                                         headers={'Content-Type': 'application/json',
+                                                  'accept': 'application/json'},
+                                         data="{}")
+
+            assert functionResp.status_code == 200
+            functionRespJson = functionResp.json()
+            assert pythonVersion == functionRespJson['majorVersion']
+
+
+    def test_functionsInZip(self):
+        """Tests if functions_deploy can handle function in zip file."""
```

Comment:
I would like to discuss zipping a little bit. I think that while our script expects already zip input, we should probably have it prepared in our data and not to try to create it during this test - it is unnecessary and we use the same package for that as the script itself for checking content that could lead to using the package badly and we would not ever know - I would suggest to just create zip by any external tool and put it into data. We should also consider if we want to zip files automatically for provided files / folders.

Reply:
I always prefer to have text files in repository rather than binary files (for easier tracking of changes). That's why the files are packed in tests. We can consider to add support for creating those zip files inside WAW (see #272)